### ISSUE TITLE
Memory Leak?

### ISSUE THREAD

**Issue by danielcrane** [gameofdimension/policy-gradient-pong#1]
First of all, thanks for this! It's a really nice re-implementation of Karpathy's original code.

The one issue I've been having is that when I run it over long periods, the memory usage builds up quite fast. After ~3 hours the memory usage is around 4GB, and after around 8 hours the memory usage was over 8GB.

It could just be me, but I was wondering if you have any idea what could be causing this?

Thanks!

**Comment by felixdae** [comment]
sorry for late response
in recent commits, i tried methods to reduce memory usage like delete intermediate object.
 but during the whole training process, memory usage still occasionally reach 6~7G.

one cause i can think is, as training going on, ai perform better, then episodes last longer, it will take more memory to store episode frames as training data. so we can reduce batch size to reduce memory usage, but i don't know if it will affect ai's performance.

besides, if you encounter abnormal exit, you dont need to restart from scratch, you can continue your training by setting variable resume to True.
